doc/event-alarm-framework/event-alarm-framework.md

```diff
@@ -0,0 +1,1265 @@
+
+Config Migration hooks will be used to persist custom profiles across an upgrade.
+
+The profile can also be applied through ztp.
+
+### 3.1.6 CLI
+The show CLI require many filters with range specifiers.
+Various filters are supported using RPC.
+
+e.g.
+```
+rpc getEventBySeqeuenceId{
+input {
+    from sequence-id;
+    to sequence-id;
+  }
+output {
+   list event-table-entries; 
+}
+```
+
+The rpc callback needs to access DB with the given set of sequence ids.
+
+The gNMI server (gnoi_client.go, gnoi.go, sonic_proto, transl_utils.go) need to be extended to support the RPC to support similar operations for gNMI.
+
+### 3.1.7 Event History Table and Current Alarm Table
+The Event History Table (EVENT) and Current Alarm List Table (ALARM) stored in EVENT_DB. 
+The size of Event History Table is 40k records or 30 days worth of events which ever hits earlier.
+A manifest file will be created with parameters to specify the number and number of days limits for
+eventd to read and enforce them.
+
+```
+root@sonic:/etc# cat eventd.json 
+{
+    "config" : {
+        "no-of-records": 40000,
+        "no-of-days": 30
+    }
+}
+```
+'no-of-records' indicates maximum number of records EVENT table can hold. The range is 1-40000.
+'no-of-days' indicates maximum number of days an event can exist in the EVENT table. The range is 1-30.
+
+When either of the limit is reached, the framework wraps around the table by discarding older records.
+
+User can send SIGINT to eventd process to force read and apply the manifest limits.
+
+An example of an event in EVENT table.
+```
+EVENT Table:
+==============================
+
+Key             : id
+
+id              : Unique sequential ID generated by the system for every event {uint64}
+type-id         : Name of the event generated {string}
+text            : Dynamic message describing the cause for the event {string}
+time-created    : Time stamp at which the event is generated {uint64}
+action          : Indicates action of the event; for one-shot events, it is empty. For alarms it could be raise, clear or acknowledge {enum}
+resource        : Object which generated the event {string}
+severity        : Severity of the event {string}
+
+127.0.0.1:6379[6]> hgetall "EVENT|1"
+ 1) "text"
+ 2) ":- handle_custom_evprofile: Custom Event Profile x.json is applied."
+ 3) "type-id"
+ 4) "CUSTOM_EVPROFILE_CHANGE"
+ 5) "id"
+ 6) "1"
+ 7) "time-created"
+ 8) "1621459327118629520"
+ 9) "resource"
+10) "/etc/evprofile/x.json"
+11) "severity"
+12) "WARNING"
+127.0.0.1:6379[6]>
+```
+
+Schema for EVENT_STATS table is as follows:
+```
+EVENT_STATS Table:
+==============================
+
+Key             : id
+
+id              : key {state}
+events          : Total events raised {uint64}
+raised          : Total alarms raised {uint64}
+cleared         : Total alarms cleared {uint64}
+acked           : Total alarms acknowledged {uint64}
```

Comment:
Why do we need for event to be acked? I thought only Alarm can be acknowledged

Reply:
Only alarms can be acknowledged but when an alarm is acknowledged or cleared or raised, it is an event that is sent by application.